components/GroupLinkSearch.js

```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+import { Combobox } from "@headlessui/react";
+import Label from "./form/Label";
+
+function GroupLinkSearch({ selectedGroup, handleGroupSelection, groups }) {
+  const [query, setQuery] = useState("");
+
+  const handleGroupChange = (e) => {
+    setQuery(e.target.value);
+    const currentValue = e.target.value;
+    if (!groups.includes(currentValue)) {
+      handleGroupSelection(currentValue);
+    }
+  };
+
+  const filteredGroup =
+    query === ""
+      ? groups.slice(0, 5)
+      : groups
+          .filter((group) => {
+            return group.toLowerCase().includes(query.toLowerCase());
+          })
+          .slice(0, 5);
+
+  return (
+    <Combobox value={selectedGroup} onChange={handleGroupSelection}>
+      <Label htmlFor="search-groups">Group</Label>
+      <Combobox.Input
+        name="search-group"
+        onChange={handleGroupChange}
+        className={`border-2 transition-all duration-250 ease-linear rounded px-6 py-2 mb-2 block w-full dark:bg-primary-high hover:border-tertiary-medium focus:ring-0 focus:border-tertiary-medium focus:outline-0`}
```

Comment:
not sure why template literal? i think we could use plain string

Reply:
I copied this from IconSearch component since they are both similar.

Reply:
In our project, we've typically used template literals when dealing with dynamic values. Since the value here seems static, a plain string might enhance readability and maintain consistency with our coding style

Reply:
Great collaboration 💪